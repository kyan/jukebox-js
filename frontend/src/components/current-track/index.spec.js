import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import keyCode from 'rc-util/lib/KeyCode'
import configureMockStore from 'redux-mock-store'
import MockTrackListJson from '__mockData__/api'
import CurrentTrack from './index'

describe('CurrentTrack', () => {
  let wrapper, track

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    describe('album', () => {
      it('renders track', () => {
        const voteMock = jest.fn()
        track = MockTrackListJson()[1].track
        track.metrics = null
        const store = configureMockStore()({
          timer: { duration: 10000, position: 8000, remaining: 700 },
          track
        })
        delete track.album.year
        wrapper = mount(
          <Provider store={store}>
            <CurrentTrack
              userID='1117795953801840xxxxx'
              track={track}
              progress={25}
              remaining={5}
              onVote={voteMock}
            />
          </Provider>
        )

        expect(wrapper).toMatchSnapshot()
        const handler = wrapper.find('.rc-slider-handle').at(1)
        wrapper.simulate('focus')
        handler.simulate('keyDown', { keyCode: keyCode.UP })
        expect(voteMock).toHaveBeenCalledWith('spotify:track:6BitwTrBfUrTdztRrQiw52', 5)
      })

      it('renders and average vote that was < 50', () => {
        const voteMock = jest.fn()
        track = MockTrackListJson()[0].track
        const store = configureMockStore()({
          timer: { duration: 10000, position: 8000, remaining: 700 },
          track
        })
        delete track.album.year
        wrapper = mount(
          <Provider store={store}>
            <CurrentTrack
              userID='1117795953801840xxxxx'
              track={track}
              progress={25}
              remaining={5}
              onVote={voteMock}
            />
          </Provider>
        )

        expect(wrapper.find('.c-nowPlaying__votingWrapper')).toMatchSnapshot()
      })

      it('renders and average vote that was 0', () => {
        const voteMock = jest.fn()
        track = MockTrackListJson()[2].track
        const store = configureMockStore()({
          timer: { duration: 10000, position: 8000, remaining: 700 },
          track
        })
        delete track.album.year
        wrapper = mount(
          <Provider store={store}>
            <CurrentTrack
              userID='1117795953801840xxxxx'
              track={track}
              progress={25}
              remaining={5}
              onVote={voteMock}
            />
          </Provider>
        )

        expect(wrapper.find('.c-nowPlaying__votingWrapper')).toMatchSnapshot()
      })
    })
  })

  describe('when no track', () => {
    it('renders nothing', () => {
      wrapper = shallow(<CurrentTrack />)

      expect(wrapper.instance()).toBeNull()
    })
  })

  describe('when no image', () => {
    it('renders default image', () => {
      track = MockTrackListJson()[0].track
      wrapper = shallow(<CurrentTrack track={track} progress={25} />)

      expect(wrapper.find('img')).toMatchSnapshot();
      expect(wrapper.find('img').prop('alt')).toMatchSnapshot()
    })
  })

  describe('when missing data', () => {
    it('renders defaults', () => {
      track = MockTrackListJson()[2].track
      delete track.addedBy
      delete track.album
      wrapper = shallow(<CurrentTrack track={track} progress={25} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
