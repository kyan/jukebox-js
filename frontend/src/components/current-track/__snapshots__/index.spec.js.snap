// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTrack render album renders track 1`] = `
<Card>
  <Image
    as="img"
    src="http://path/to/cool/image.png"
    ui={true}
  />
  <CardContent>
    <div
      className="progress-container"
    >
      <span
        className="remaining-text"
      >
        NaN:NaN
      </span>
      <span
        className="track-length"
      >
        4:06
      </span>
      <Progress
        className=""
        percent={25}
        prefixCls="rc-progress"
        strokeColor="#2db7f5"
        strokeLinecap="round"
        strokeWidth={1}
        style={Object {}}
        trailColor="#D9D9D9"
        trailWidth={1}
      />
    </div>
    <CardHeader>
      No Time for Caution
    </CardHeader>
    <CardMeta>
      Hans Zimmer
    </CardMeta>
    <CardDescription>
      Interstellar: Original Motion Picture Soundtrack (Deluxe Digital Version)
       (2014)
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <AddedBy
      users={
        Array [
          Object {
            "_id": "123",
            "addedAt": "2019-12-17T13:11:37.316Z",
            "fullname": "Big Rainbowhead",
          },
        ]
      }
    />
  </CardContent>
</Card>
`;

exports[`CurrentTrack render composer renders track 1`] = `
<Card>
  <Image
    as="img"
    src="default-artwork.png"
    ui={true}
  />
  <CardContent>
    <div
      className="progress-container"
    >
      <span
        className="remaining-text"
      >
        NaN:NaN
      </span>
      <span
        className="track-length"
      >
        2:04
      </span>
      <Progress
        className=""
        percent={25}
        prefixCls="rc-progress"
        strokeColor="#2db7f5"
        strokeLinecap="round"
        strokeWidth={1}
        style={Object {}}
        trailColor="#D9D9D9"
        trailWidth={1}
      />
    </div>
    <CardHeader>
      Silent Running
    </CardHeader>
    <CardMeta>
      Joan Baez
    </CardMeta>
    <CardDescription>
      Peter Schickele
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <AddedBy />
  </CardContent>
</Card>
`;
